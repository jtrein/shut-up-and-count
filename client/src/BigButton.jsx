import React from 'react';
import { bool, func } from 'prop-types';

import './button.css';
import styles from './BigButton.styles';

const BigButton = ({ disabled = false, onClick }) => (
  <div style={styles.wrap}>
    <button disabled={disabled} onClick={onClick}>Count the words.</button>
  </div>
);

BigButton.defaultProps = {
  disabled: false,
};

BigButton.propTypes = {
  disabled: bool,
  onClick: func.isRequired,
};

export default BigButton;
