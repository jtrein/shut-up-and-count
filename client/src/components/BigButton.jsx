import React from 'react';
import { bool, func } from 'prop-types';

import '../styles/button.css';
import styles from '../styles/BigButton.styles';

const BigButton = ({ disabled = false, onClick }) => (
  <div style={styles.wrap}>
    <button disabled={disabled} onClick={onClick}>Count the words.</button>
  </div>
);

BigButton.defaultProps = {
  disabled: false,
};

BigButton.propTypes = {
  disabled: bool,
  onClick: func.isRequired,
};

export default BigButton;
