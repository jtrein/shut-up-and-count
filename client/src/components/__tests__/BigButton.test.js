/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BigButton from '../BigButton';

Enzyme.configure({ adapter: new Adapter() });

test('renders without crashing', () => {
  const div = document.createElement('div');
  render(<BigButton onClick={() => {}} />, div);
  unmountComponentAtNode(div);
});

test('renders expected initial components', () => {
  const wrapper = shallow(<BigButton onClick={() => {}} />);
  expect(wrapper.find('div').length).toEqual(1);
  expect(wrapper.find('button').length).toEqual(1);
});

test('button is initially disabled', () => {
  const wrapper = shallow(<BigButton onClick={() => {}} />);
  expect(wrapper.find('button').props().disabled).toEqual(false);
});

test('button has passed onClick func', () => {
  const wrapper = shallow(<BigButton onClick={() => {}} />);
  expect(wrapper.find('button').props().onClick.toString()).toEqual('() => {}');
});

test('button calls onClick once when clicked', () => {
  const spy = jest.fn();
  const wrapper = shallow(<BigButton onClick={spy} />);
  wrapper.find('button').simulate('click');
  expect(spy.mock.calls.length).toEqual(1);
});
