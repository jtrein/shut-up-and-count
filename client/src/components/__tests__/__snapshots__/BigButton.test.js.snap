// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: renders as expected 1`] = `
<div
  style={
    Object {
      "marginTop": 24,
    }
  }
>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Count the words.
  </button>
</div>
`;
